### DISCUSSION TITLE
Pure Javascript

### DISCUSSION THREAD

**Discussion by applibs** [zalify/easy-email-editor#281] [General]
Could someone please show how to use easy email in pure Javascript?
I've tried some tests, but nothing works.

**Comment by Kaleb11** [comment]
Easy email sending in pure JavaScript is not possible as JavaScript runs in a user's browser and does not have the capability to send emails directly. However, you can use JavaScript to send emails through a third-party email service that provides an API.

One such popular email service is SendGrid, which offers an API that allows you to send emails programmatically. Here's an example of how you can use SendGrid's API to send an email using JavaScript:

`const sgMail = require('@sendgrid/mail');
sgMail.setApiKey(process.env.SENDGRID_API_KEY);

const msg = {
  to: 'recipient@example.com',
  from: 'sender@example.com',
  subject: 'Sending email with SendGrid',
  text: 'This email was sent using SendGrid!'
};

sgMail.send(msg)
  .then(() => console.log('Email sent'))
  .catch((error) => console.error(error));
`

**Comment by dev-alihasan** [comment]
**First, you need to include the EasyEmail library in your HTML file**
`<script src="https://cdn.jsdelivr.net/npm/easyemail@1.0.0/dist/easyemail.min.js"></script>
`
**next, you can use the EasyEmail object to send an email using your SMTP server.**
`// Initialize EasyEmail with your SMTP credentials
const email = new EasyEmail({
  host: 'smtp.example.com',
  port: 587,
  secure: false,
  auth: {
    user: 'your_username',
    pass: 'your_password'
  }
});

// Send the email
email.send({
  to: 'recipient@example.com',
  subject: 'Test Email',
  text: 'This is a test email sent using EasyEmail!'
})
  .then(() => {
    console.log('Email sent successfully!');
  })
  .catch((err) => {
    console.error('Error sending email:', err);
  });
`

**Comment by applibs** [comment]
I don't want to send emails, just use it as a tool.

**Comment by applibs** [comment]
Now its working well after some experiments.

**Comment by sufianmehmood** [comment]
@applibs can you share the pure javascript code here as i have a similar use case and considering this as a email editor, currently using CKEditor it works but not really happy with the formatting capabilities, considering to use this as the editor to retrieve the html

**Reply by applibs** [reply]
I already fixed my problem. Thank you.

**Reply by sufianmehmood** [reply]
can you please share the code or sample here?
